### ISSUE TITLE
ES6: Create a lot of instances: error Object doesn't support property or method

### ISSUE THREAD

**Issue by alexzhaosheng** [Microsoft/ChakraCore#2532]
I've noticed there's a similar issue has been reported here: https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/6664966/, but it's marked "NOT REPRODUCIBLE".  Therefore I'd like to report it here with more detailed information:

I have a class like this:

```
class Point{
    constructor(x = 0, y = 0) {
            this.x = Number(x);
            this.y = Number(y);
        }
    add(point){...}
    subtract(point){...}
}
```

We have a really complex system, therefore there will be thousand of Point instance, and after a lot of instances are created, new Point() will not return the new instance, instead, it returns the class declaration it-self. 
Here's my test code:
```
var p = new Point(0,0);
if(!p.add){
    if(p == Point)
        console.write("boom!");
}
```
And yes, I get quite a few "boom!" in Edge. The same code works fine in the other browsers.

Here are what I can tell about this error:
1. it happens with any class that comes with large number of instance when the browser is in heavy loads
2. when it happens, "new" keyword returns the class definition, not the instance
3. when debug from visual studio 2015, I tried to re-run the code by dragging the executing point when it stops on exception, then it just works

I may deploy my system to internet in 1 month, if you can't reproduce it, please let me known, this error is 100% reproduceable with our system.

**Comment by ianwjhalliday** [comment]
@suwc this sounds familiar. Was this a JIT bug that was fixed a while back?

@alexzhaosheng I cannot repro in the console using ch.exe.  Is there more code to your repro? This is what I tried:
```javascript
class Point{
    constructor(x = 0, y = 0) {
            this.x = Number(x);
            this.y = Number(y);
        }
    add(point){}
    subtract(point){}
}

function f() {
    var p = new Point(0, 0);
    if (!p.add) {
        if (p === Point) {
            print("Boom!")
        }
    }
}

for (let i = 0; i < 1000000; i++) {
    f();
}
```

**Comment by suwc** [comment]
This appears to be duplicate of the issues that have been fixed:

(back in Sept. '16) ES6 constructor returns class instead of object
https://github.com/Microsoft/ChakraCore/issues/1496

and

(2/14/17) ES6 constructor returns class instead of object while switching F12 on or off
https://github.com/Microsoft/ChakraCore/issues/2499

@alexzhaosheng Please advise if F12 is involved in your repro case. Either way I would be very interested to see it. Thanks for reporting!

**Comment by alexzhaosheng** [comment]
@ianwjhalliday  I've also tried the similar testing code and failed to reproduce. It only happens inside our system. Our system is a complex SVG based engineering drawing platform which uses quite a lot resources, I assume this bug only happens when browser is under heavy load

@suwc  It happens no matter F12 is pressed or not.  And it happens when visual studio debugger is attached with Edge

**Comment by suwc** [comment]
@alexzhaosheng Based on your comment "And it happens (*only*?) when visual studio debugger is attached with Edge", this looks like https://github.com/Microsoft/ChakraCore/issues/2499, which was fixed only a couple of days ago.

**Comment by alexzhaosheng** [comment]
@suwc  Sorry I didn't make myself clear.
This problem happens when:
1.Open the site in Edge without develop tools window and Visual studio debugger (pressing F12)
2.Open the site in Edge with develop tools window
3.Open the site in Edge when Visual studio debugger is attached

So basically it happens in all situations.

I'd be very happy to do the test again after #2499 being released, could you tell me when it will be released?

**Comment by suwc** [comment]
@alexzhaosheng Sorry I cannot comment on release dates. (Feel free to search "Windows 10 Creators Update" though)
Could you share your Edge version number?
If it is 38.14393.0.0 or something like that, that is "Anniversary Update", in which #1496 and #2499 are expected to repro.

**Comment by alexzhaosheng** [comment]
@suwc  Thanks for the reply, my Edge version is 38.14393.0.0

**Comment by ondras** [comment]
I believe I have the problematic behavior reproducible on this fiddle: http://jsfiddle.net/0k59gbL8/5/ . It happens both with and without devtool open.

@suwc can you please confirm that the link above exhibits the bug reported in this issue?

**Comment by suwc** [comment]
@ondras thanks for the test case. Yes this is the issue mentioned above that will be fixed in "Creators Update".

**Comment by ondras** [comment]
@suwc thank you very much for confirming this! 👍
